templates/viewer.php

```diff
@@ -282,13 +283,13 @@
                   <span data-l10n-id="print_label">Print</span>
                 </button>
 
-                <button id="download" class="toolbarButton hiddenMediumView" title="Save" tabindex="33" data-l10n-id="save" hidden="true">
+                <button id="download" class="toolbarButton hiddenMediumView" title="Save" tabindex="33" data-l10n-id="save">
```

Comment:
The _Save_ button should not be enabled if there are no annotations. Internally `PDFViewerApplication` checks if [`this.pdfDocument?.annotationStorage.size > 0`](https://github.com/mozilla/pdf.js/blob/v3.9.179/web/app.js#L1186-L1192) to either download or save (that is, apply the annotations and then download) the file, so in an ideal world the Vue component could watch `PDFViewerApplication.pdfDocument?.annotationStorage.size` and enable or disable the button depending on the case.

Unfortunately we do not live in an ideal world :-( `annotationStorage.size` does not seem to be reactive (probably because it is [a getter that depends on a private field](https://github.com/mozilla/pdf.js/blob/v3.9.179/src/display/annotation_storage.js#L145-L147), although I am not fully sure if that is the actual problem), so it can not be watched for changes.

But there is an alternative (a bit hacky, though :shrug:). The annotation storage provides callbacks [so the PDFViewerApplication can know if the storage was modified](https://github.com/mozilla/pdf.js/blob/v3.9.179/web/app.js#L1743-L1756) (even if it ends directly checking the size in `downloadOrSave` rather than `this._annotationStorageModified`), so `onSetModified` could be used to react when the storage is modified. However, as it is a single callback and not an event to add listeners to the original callback needs to be wrapped to be able to still call it while also changing the button state.

This approach is not perfect either, as `onSetModified` seems to be called when just selecting the tools to add an annotation even if no annotation ends being added, but this seems to be the best that can be done (or, at least, that I could find :-) ).

In any case, if the button is not enabled when there are no annotations then it would not be necessary to show a message if trying to save in that case (as it would not be possible to do it).